src/Configurator.php

```diff
@@ -112,14 +112,20 @@ public function initFields(EntitySchema $entity, \ReflectionClass $class, string
             }
 
             $field = $this->initField($property->getName(), $column, $class, $columnPrefix);
-            $field->setEntityClass($property->getDeclaringClass()->getName());
+            $field->setEntityClass($this->findOwningEntity($class, $property->getDeclaringClass())->getName());
             $entity->getFields()->set($property->getName(), $field);
         }
     }
 
     public function initRelations(EntitySchema $entity, \ReflectionClass $class): void
     {
         foreach ($class->getProperties() as $property) {
+            // ignore properties declared by parent entties
+            // otherwise all the relation columns declared in parent would be duplicated across all child tables in JTI
+            if ($this->findOwningEntity($class, $property->getDeclaringClass())->getName() !== $class->getName()) {
+                continue;
+            }
```

Comment:
I've added tests for this case and there is an issue when an entity extended from another entity without STI or JTI.

In thos case all the relations of the parent entity have to be cloned into the child class.

Reply:
There is `TableInheritance` generator. It might be better to clean unnecessary relations there in cases STI or JTI

Reply:
Thanks for the review.

You are right, it is quite a nasty use case though. I will take a look at it in the evening, 
It makes sense to clear the entity in the inheritance generator, I will try to move the logic there.

The column ownership use cases are alright?